src/typegoose.ts

```diff
@@ -161,6 +161,11 @@ export function buildSchema<T, U extends AnyParamConstructor<T>>(cl: U, options?
 
       break;
     }
+
+    if (parentCtor.name === cl.name) {
```

Comment:
you dont need to add this check, because the while loop acts TOPDOWN, and because of `Object.getPrototypeOf(cl.prototype).constructor` it starts with the class that is extended from

this check will *only* trigger if the class that is extended has the same name, which would skip that class, and that shouldnt happen

Reply:
Sorry, apparently I was too much focused on solving my problem. You're right. I'll go back to my initial proposal, to check function equality for hooks and plugins.